Projects detail window

inakiesk opened the pull request:
This branch is dedicated to developing a window with detailed information about every project, the features are the following:
-A popup window is generated when the user clicks on "See project" button.
-The popup window has the corresponding information for every section.
-The popup windows close when you press the "X" icon.
-The popup windows have been adapted to work properly for mobile and desktop version.
-An object is used to store the information of all popup windows, when the user clicks on the "See project" button, the information is then selected depending of the work section from the object and displayed correctly, saving some lines of code.
-Made sure to make image sizes, buttons, etc, as close as possible to Figma template.

inakiesk committed:
Modify image properties for better display

inakiesk committed:
Organize main code for work window display

inakiesk committed:
Fix card positioning issues

inakiesk committed:
Change button size to make sure it follows figma design

inakiesk committed:
Fix image positioning

Ugboaja-Uchechi commented on index.js:
```diff
@@ -32,56 +32,63 @@ const buttonWork = document.querySelector('.button-work');
-  clonedDesktopImage.classList.add('cloned-desk-img');
   const popInfoContainer = document.createElement('div');
   const popBodyContainer = document.createElement('div');
+  const description = document.createElement('p');
+  const clonedTech = popUp.technologies[f].cloneNode(true);
+  const githubTech = document.createElement('li');
```
- [ ] Kindly make sure you use an array of objects instead of an object to display your data as specified by the previous code reviewer 🙏 
- [ ] Kindly make sure you populate every property in your object. For example, the `image` property in your object is just a string please make sure you add the `image` instead 🙏

inakiesk replied:
Hello @Ugboaja-Uchechi !
First of all, thank you for your review!
Before I continue I would just like to see what you think about the change that I did to see if you would want me to change the code more.
For making my object an array of objects I did this: 
![popup-code](https://user-images.githubusercontent.com/93748135/180490958-e5d7f824-8e7a-432f-bde7-52dfa2722db8.png)

This way, the main object technically has other objects inside, they look empty at the moment, but I have this line of code to fill it:

![popup-selector](https://user-images.githubusercontent.com/93748135/180491133-865e59ee-3954-41b4-9351-ead79172f90e.png)

That way the objects inside the main object have an array of all the elements I will need to display the work section, now I can select the ones I need for each section with the indexes by using this:
![popup-index](https://user-images.githubusercontent.com/93748135/180492817-28b0252a-4a4d-4f44-ac92-84f9add5855d.png)

So the image object should not be empty now.
I did a console log to make sure they were being added:
![popup-group](https://user-images.githubusercontent.com/93748135/180491402-4da5cc0b-0a20-4d21-b110-14ff2c7ad8cf.png)

I decided to do the main object like this because this makes my code easy to adapt, and in case that another work section is added, I would not have to create another object, the elements will be added automatically to the array, and the next section will be displayed correctly without having to add any code at all on the Javascript file!

What do you think about this? Do you think it is correct by having the code like this or do you think it would be best if I changed the object that I made completely? Changing my object that gets autofilled to another object that already has the information would mean changing a lot of my code structure but I want to make sure I have a code you would feel comfortable accepting. So please let me know if this meets your requirements or if you want me to do a more radical change on the objects.
Thank you!

Ugboaja-Uchechi reviewed (changes requested):
Hi @inakiesk, 

**Note: Please make sure you don't delete or merge a pull request if it hasn't been approved. If you do so by mistake please let the next code reviewer know** 🙏 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- Linters are passing ✔️ 
- README is descriptive 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

bimbolabuari reviewed (commented):
...

bimbolabuari reviewed (commented):
...

bimbolabuari reviewed (commented):
Hello @inakiesk  👋,

_This is an additional review as a result of the second opinion request._

- The changes requested by the previous CR were right it won't downgrade your code but make it DRY(Don't repeat yourself)
  - The first request change was to use an array of objects, Therefore, you need to create an array with objects
  For examples:
```
const popUp = {
  workHeader: '',
  subHeader: '',
  image: '',
  description:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.",
  technologies: '',
  linkLive: 'See live',
  linkSource: 'See source',
};
const anotherPopup = {
  workHeader: '',
  subHeader: '',
  image: '',
  description:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.",
  technologies: '',
  linkLive: 'See live',
  linkSource: 'See source',
};
const popupArray = [popUp, anotherPopup]  // This is an array of objects
```

You can also use this method:

```
const popUp = [
{
  workHeader: '',
  subHeader: '',
  image: '',
  description:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.",
  technologies: '',
  linkLive: 'See live',
  linkSource: 'See source',
}

{
  workHeader: '',
  subHeader: '',
  image: '',
  description:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.",
  technologies: '',
  linkLive: 'See live',
  linkSource: 'See source',
}
]

```

- For the second request changes, Kindly add your object value instead of just an empty string, this will make your code cleaner and professional and also let you render the HTML elements dynamically using javascript instead of getting the elements with DOM manipulation, In order words the HTML elements text should be your object value 👍 

https://github.com/inakiesk/Portfolio-mobile-version/blob/7d00bc0d5e4b3101651689994a8e4ec320de138f/index.js#L15-L105

**More insight:** Kindly render the work section dynamically with an array of objects using the required number of projects, then add the object value and just use a unique ID to get the popup details from the array of objects.

Keep rocking 🚀!

inakiesk committed:
Did basic structure of new code

inakiesk committed:
Complete and organize function structure for correct display

inakiesk commented:
Summary of new changes:
-Deleted previous code and replaced the auto generated object with a group of objects that will correctly display the object information. There are no more empty objects.
-Added a feature to the "See live" and "See source" button. It will now properly direct you to the corresponding website on click.

Hope you like the new changes!

mwanawabangona reviewed (approved):
Hi @inakiesk ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  
![](https://camo.githubusercontent.com/d348d0cdaa16b202354d12402e048f52cd4d589b4ec0e55f07d62c2f22753376/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f336f373246634a6d4c7a4964594a646d44652f67697068792e676966)

## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implimenting the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
